jiant/tasks/lib/bucc2018.py

```diff
@@ -0,0 +1,424 @@
+import hashlib
+import numpy as np
+import torch
+from dataclasses import dataclass
+from typing import List
+
+from jiant.tasks.core import (
+    BaseExample,
+    BaseTokenizedExample,
+    BaseDataRow,
+    BatchMixin,
+    Task,
+    TaskTypes,
+)
+from jiant.tasks.lib.templates.shared import (
+    construct_single_input_tokens_and_segment_ids,
+    create_input_set_from_tokens_and_segments,
+)
+from jiant.utils.python.io import read_file, read_file_lines
+
+
+@dataclass
+class Example(BaseExample):
+    guid: str
+    is_english: bool
+    text: str
+    text_hash: str
+
+    def tokenize(self, tokenizer):
+        return TokenizedExample(
+            guid=self.guid,
+            is_english=self.is_english,
+            text_tokens=tokenizer.tokenize(self.text),
+            text_hash=self.text_hash,
+        )
+
+
+@dataclass
+class TokenizedExample(BaseTokenizedExample):
+    guid: str
+    is_english: bool
+    text_tokens: List
+    text_hash: str
+
+    def featurize(self, tokenizer, feat_spec):
+        unpadded_inputs = construct_single_input_tokens_and_segment_ids(
+            input_tokens=self.text_tokens, tokenizer=tokenizer, feat_spec=feat_spec,
+        )
+        input_set = create_input_set_from_tokens_and_segments(
+            unpadded_tokens=unpadded_inputs.unpadded_tokens,
+            unpadded_segment_ids=unpadded_inputs.unpadded_segment_ids,
+            tokenizer=tokenizer,
+            feat_spec=feat_spec,
+        )
+        return DataRow(
+            guid=self.guid,
+            input_ids=np.array(input_set.input_ids),
+            input_mask=np.array(input_set.input_mask),
+            segment_ids=np.array(input_set.segment_ids),
+            is_english=self.is_english,
+            tokens=unpadded_inputs.unpadded_tokens,
+            text_hash=self.text_hash,
+        )
+
+
+@dataclass
+class DataRow(BaseDataRow):
+    guid: str
+    input_ids: np.ndarray
+    input_mask: np.ndarray
+    segment_ids: np.ndarray
+    is_english: bool
+    tokens: list
+    text_hash: str
+
+
+@dataclass
+class Batch(BatchMixin):
+    input_ids: torch.LongTensor
+    input_mask: torch.LongTensor
+    segment_ids: torch.LongTensor
+    is_english: torch.BoolTensor
+    tokens: list
+    text_hash: list
+    guid: list
+
+
+class Bucc2018Task(Task):
+    Example = Example
+    TokenizedExample = Example
+    DataRow = DataRow
+    Batch = Batch
+
+    TASK_TYPE = TaskTypes.EMBEDDING
+
+    def __init__(self, name, path_dict, language):
+        super().__init__(name=name, path_dict=path_dict)
+        self.language = language
+
+    def get_train_examples(self):
+        raise NotImplementedError()
```

Comment:
I'm being pedantic but is this method just not implemented yet or should it never be implemented? If it is the latter, let's change it to a `RuntimeError()`

Reply:
Will update.